### DISCUSSION TITLE
Run mkdocs gh-deploy --force exit code 1

### DISCUSSION THREAD

**Discussion by meroxdotdev** [squidfunk/mkdocs-material#6853] [Q&A] [answered]
Config value 'plugins': The "git-revision-date-localized" plugin is not installed
Aborted with 1 conifugration errors!
Error: process completed with exit code 1

I'm not sure how to resolve this one
![githuberr](https://github.com/squidfunk/mkdocs-material/assets/35839230/3c6b8de9-80e6-4deb-a2f4-d9ed51d6e2a9)

This is the section from my mkdocs.yml:
![codegithub](https://github.com/squidfunk/mkdocs-material/assets/35839230/646ea2c4-6263-456d-93ee-01af78b40e55)

 P.S: 
If I remove:
  - git-revision-date-localized:
      enable_creation_date: true
  - social
  
From plugins in mkdocs.yml, everything work as expected on deployment with ci.yml via github-pages

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @mer0x,
if you're talking about https://github.com/mer0x/merox.cloud then you forgot to install the `git-revision-date-localized` plugin in your CI.yml workflow.

You only install `mkdocs-material`:
https://github.com/mer0x/merox.cloud/blob/42b21a978c189a9904862740ac716f44e9798585/.github/workflows/ci.yml#L28

**Reply by kamilkrzyskow** [reply]
~~I'm not sure where are the other warnings coming from 🤔~~ EDIT: read next comment
Apart of that I noticed that you pushed the `.cache` directory and some other config stuff to the repository as you don't have a proper `.gitignore` file set up. I think you would need to first do a commit to remove them and later add the `.gitignore` for Git to handle the exclusions.

**Reply by kamilkrzyskow** [reply]
You've put the `custom_dir` inside the `docs` directory, this is an anti-pattern of sort. This shouldn't be the case, as the `home.html` file will still "be" rendered.
https://merox.cloud/assets/home.html
As per the customization guide put it outside of the docs directory next to `mkdocs.yml`: 
https://squidfunk.github.io/mkdocs-material/customization/#setup-and-theme-structure

The warning about the unrecognized config options are coming from the `mkdocs.yml` file:
https://github.com/mer0x/merox.cloud/blob/ea0e1cea5e0461ed7e8a7cc48d893bebfede18e0/mkdocs.yml#L208
you can't set them like that, you should put everything under the `extra` config options:
https://www.mkdocs.org/user-guide/configuration/#extra
and here you should use `{{ config.extra.custom_name }}`
https://github.com/mer0x/merox.cloud/blob/ea0e1cea5e0461ed7e8a7cc48d893bebfede18e0/docs/assets/home.html#L272

**Reply by meroxdotdev** [reply]
Thank you so much for your help, really appreciate it!

**Reply by meroxdotdev** [reply]
I've corrected the location of the assets and mkdocs.yml, but now I can't see anymore the homepage, it's blank, can you take a look ? Most probably I miss something

**Reply by kamilkrzyskow** [reply]
`view-source:https://merox.cloud/` shows the contents of the `home.html`, it has the `tx-container` class etc. 
However, you forgot to switch from `config.custom_name` to `config.extra.custom_name` for the extra data to be found in the template.

There are other things that show warnings in the github action log, like the fetch depth for the git revision plugin:
https://timvink.github.io/mkdocs-git-revision-date-localized-plugin/#note-when-using-build-environments

**Reply by kamilkrzyskow** [reply]
I also think it would be easier to use `mkdocs serve` on your local system as this would allow for faster debugging, and separate the published and development site.
Now it looks like you make a change and deploy it to see how it looks, which is a bit wasteful 🤔

**Reply by meroxdotdev** [reply]
My bad :)), Yes defnitely this way I will do it form now, I started to add .extra, but the gradient background and buttons still doesn't appear

**Reply by kamilkrzyskow** [reply]
https://github.com/mer0x/merox.cloud/blob/master/assets/home.html
There are 12 matches for `config.`, but only 8 matches for `config.extra` and 4 `config.home` 😉 
Not all of them were changed...

**Reply by meroxdotdev** [reply]
> https://github.com/mer0x/merox.cloud/blob/master/assets/home.html
> 
> There are 12 matches for `config.`, but only 8 matches for `config.extra` and 4 `config.home` 😉 
> 
> Not all of them were changed...

Now all loks changed and everything looks better, but the gradient background still doesnt appear

**Reply by kamilkrzyskow** [reply]
I don't have time to debug it more thoroughly, but a quick inspection indicates an invalid CSS, which prevents all of it from being processed, the issue with the invalid value for `height` is probably a false-positive, as all of them were fine. But the `;` should be removed from the gradient.

https://github.com/squidfunk/mkdocs-material/assets/34622465/6388267e-20bb-43f9-9a6f-5c25c35dd1d7
